### ISSUE TITLE
Overflow is not working

### ISSUE THREAD

**Issue by George-Miao** [DioxusLabs/dioxus#371]
**Problem**

Long text is not hidden or trimmed properly according to `overflow` attribute, which will cause out-of-boundry panic.

**Steps To Reproduce**

Code:

```rust
fn main() {
    dioxus_tui::launch(|cx| {
        cx.render(rsx!{
            div {
                "overflow": "hidden",
                "------------------------------------------------------------------------------------------------------------------"
            }
        })
    });
}
```

**Expected behavior**

Text should be trimmed or wrapped when length is greater than container's width.

**Screenshots**

![image](https://user-images.githubusercontent.com/38549822/165022309-fbb48e33-782b-429f-a48d-23c85f9e6ce4.png)

**Environment:**
 - Dioxus version: [`0.2.3`]
 - Rust version: [`rustc 1.61.0-nightly (8d60bf427 2022-03-19)`]
 - OS info: [`Linux`]
 - Terminal: [`Alacritty`]
 - App platform: [`tui`]

**Questionnaire**

- [X] I'm interested in fixing this myself but don't know where to start
- [ ] I would like to fix and I have a solution
- [ ] I don't have time to fix this right now, but maybe later

**Comment by Demonthos** [comment]
Good catch. It looks like the text rendering in https://github.com/DioxusLabs/dioxus/blob/master/packages/tui/src/render.rs#L66 ignores the size of the layout right now. Stretch2 handles overflow, we just need to trim the text if it goes out of bounds. I think the width might need to be explicitly set to 100% for the div right now, but that is a bug in stretch2?
